dev-games/aseprite/aseprite-1.3.8.1.ebuild

```diff
@@ -0,0 +1,235 @@
+}
+
+src_configure() {
+	# -Werror=strict-aliasing, -Werror=odr, -Werror=lto-type-mismatch
+	# https://bugs.gentoo.org/924692
+	# https://github.com/aseprite/aseprite/issues/4413
+	#
+	# There is a bundled skia that fails with ODR errors. When excluding just
+	# skia from testing, aseprite itself failed with strict-aliasing (before
+	# upstream PR#84), and when that is disabled, fails again with ODR and
+	# lto-type-mismatch issues.
+	#
+	# There are a lot of issues, so don't trust any fixes without thorough
+	# testing.
+	filter-lto
+
+	einfo "Skia configuration"
+	cd "${WORKDIR}/skia-${SKIA_REV}" || die
+
+	tc-export AR CC CXX
+
+	passflags() {
+		local _f _x
+		_f=( ${1} )
+		_x="[$(printf '"%s", ' "${_f[@]}")]"
+		myconf_gn+=( ${2}="${_x}" )
+	}
+
+	local myconf_gn=(
+		ar=\"${AR}\"
+		cc=\"${CC}\"
+		cxx=\"${CXX}\"
+
+		is_official_build=true
+		is_component_build=false
+		is_debug=false
+
+		skia_use_egl=false
+		skia_use_dawn=false
+		skia_use_dng_sdk=false
+		skia_use_metal=false
+		skia_use_sfntly=false
+		skia_use_wuffs=false
+
+		skia_enable_pdf=false
+		skia_enable_svg=false
+		skia_use_expat=false
+		skia_use_ffmpeg=false
+		skia_use_fontconfig=false
+		skia_use_freetype=true
+		skia_use_gl=true
+		skia_use_harfbuzz=true
+		skia_use_icu=false
+		skia_use_libjpeg_turbo_decode=true
+		skia_use_libjpeg_turbo_encode=true
+		skia_use_libpng_decode=true
+		skia_use_libpng_encode=true
+		skia_use_libwebp_decode=$(usex webp true false)
+		skia_use_libwebp_encode=$(usex webp true false)
+		skia_use_lua=false
+		skia_use_vulkan=false
+		skia_use_x11=false
+		skia_use_xps=false
+		skia_use_zlib=true
+	)
+
+	passflags "${CFLAGS}" extra_cflags_c
+	passflags "${CXXFLAGS}" extra_cflags_cc
+	passflags "${LDFLAGS}" extra_ldflags
+	myconf_gn="${myconf_gn[@]}"
+	set -- gn gen --args="${myconf_gn% }" out/Static
+	echo "$@"
+	"$@" || die
+
+	einfo "Aseprite configuration"
+	cd "${WORKDIR}" || die
+
+	local mycmakeargs=(
+		-DENABLE_CCACHE=OFF
+		-DENABLE_DESKTOP_INTEGRATION=ON
+		-DENABLE_STEAM=OFF
+		-DENABLE_TESTS="$(usex test)"
+		-DENABLE_QT_THUMBNAILER="$(usex kde)"
+		-DENABLE_UPDATER=OFF
+		-DENABLE_UI=ON
+		-DENABLE_WEBP="$(usex webp)"
+		-DLAF_WITH_EXAMPLES=OFF
+		-DLAF_WITH_TESTS="$(usex test)"
+		-DFULLSCREEN_PLATFORM=ON
+		-DSKIA_DIR="${WORKDIR}/skia-${SKIA_REV}/"
+		-DSKIA_LIBRARY_DIR="${WORKDIR}/skia-${SKIA_REV}/out/Static/"
+		-DSKIA_LIBRARY="${WORKDIR}/skia-${SKIA_REV}/out/Static/libskia.a"
+		-DSKSHAPER_LIBRARY="${WORKDIR}/skia-${SKIA_REV}/out/Static/libskshaper.a"
+		-DUSE_SHARED_CMARK=ON
+		-DUSE_SHARED_CURL=ON
+		-DUSE_SHARED_FMT=ON
+		-DUSE_SHARED_FREETYPE=ON
+		-DUSE_SHARED_GIFLIB=ON
+		-DUSE_SHARED_HARFBUZZ=ON
+		-DUSE_SHARED_JPEGLIB=ON
+		-DUSE_SHARED_JSON11=OFF		# Custom methods added to bundled version
+		-DUSE_SHARED_LIBARCHIVE=ON
+		-DUSE_SHARED_LIBPNG=ON
+		-DUSE_SHARED_PIXMAN=ON
+		#-DUSE_SHARED_TINYXML=ON
+		-DUSE_SHARED_WEBP=ON
+		-DUSE_SHARED_ZLIB=ON
+		#-DFETCHCONTENT_FULLY_DISCONNECTED=ON # Fix bug 935448
```

Comment:
```diff
commit f176dd7bfbc6da9df1381ab2c4692ac47aa307d0
Author:     m007x <151087513+m007x@users.noreply.github.com>
AuthorDate: Fri Sep 20 08:12:17 2024
Commit:     GitHub <noreply@github.com>
CommitDate: Fri Sep 20 08:12:17 2024

    Update aseprite ebuild to 1.3.8.1
    
    Signed-off-by: m007x <151087513+m007x@users.noreply.github.com>

diff --git a/dev-games/aseprite/aseprite-1.3.5.ebuild b/dev-games/aseprite/aseprite-1.3.8.1.ebuild
similarity index 95%
copy from dev-games/aseprite/aseprite-1.3.5.ebuild
copy to dev-games/aseprite/aseprite-1.3.8.1.ebuild
index 8717c09abd5e..81344bbf90b6 100644
--- a/dev-games/aseprite/aseprite-1.3.5.ebuild
+++ b/dev-games/aseprite/aseprite-1.3.8.1.ebuild
@@ -72,13 +72,14 @@ S="${WORKDIR}"
 
 PATCHES=(
 	"${FILESDIR}/skia-${SKIA_VER}_remove_angle2.patch"
-	"${FILESDIR}/${PN}-1.2.40_shared_libarchive.patch"
+	# "${FILESDIR}/${PN}-1.2.40_shared_libarchive.patch"
+	"${FILESDIR}/${PN}-1.3.8.1_shared_lib_archive.patch"
 	"${FILESDIR}/${PN}-1.3.2_shared_json11.patch"
 	"${FILESDIR}/${PN}-1.3.2_shared_webp.patch"
 	"${FILESDIR}/${PN}-1.2.35_laf_fixes.patch"
 	"${FILESDIR}/${PN}-1.3.2_shared_fmt.patch"
 	"${FILESDIR}/${PN}-1.3.2_strict-aliasing.patch"
-	"${FILESDIR}"/aseprite-1.3.5_laf-strict-aliasing.patch
+	#"${FILESDIR}"/aseprite-1.3.5_laf-strict-aliasing.patch
 )
 
 src_prepare() {
@@ -198,9 +199,10 @@ src_configure() {
 		-DUSE_SHARED_LIBARCHIVE=ON
 		-DUSE_SHARED_LIBPNG=ON
 		-DUSE_SHARED_PIXMAN=ON
-		-DUSE_SHARED_TINYXML=ON
+		#-DUSE_SHARED_TINYXML=ON
 		-DUSE_SHARED_WEBP=ON
 		-DUSE_SHARED_ZLIB=ON
+		#-DFETCHCONTENT_FULLY_DISCONNECTED=ON # Fix bug 935448
 	)
 	cmake_src_configure
 }
```

This doesn't have a good explanation for why you're making these changes. And actually, you should not be disabling system dependencies... also  it's not obvious why you added *new* lines which are commented out.

The bug you link to already has an open PR that tries to solve it, so it may be worth looking at what that ebuild does. :)